### PR TITLE
Supplier assembly email

### PR DESCRIPTION
**Pull request by andrewfraser73**
Sends a simple text email from a button on the purchase order page to Logistics with details of any supplier assembly so they can kit it.

it shows the components required (mapped from bom_Details_view). SUKIT sort of does this for them but sometimes Logistics do it manually if they see several orders that require shared components.

### CODE REVIEW

File: src/Domain.LinnApps/PurchaseOrders/ISupplierKitService.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+﻿namespace Linn.Purchasing.Domain.LinnApps.PurchaseOrders
+{
+    using System.Collections.Generic;
+
+    public interface ISupplierKitService
+    {
+        public IEnumerable<SupplierKit> GetSupplierKits(PurchaseOrder order, bool getComponents = false);
```
Comment (lewisrenfrew) [inline src/Domain.LinnApps/PurchaseOrders/ISupplierKitService.cs:7]: no need for access modifiers in interfaces (I still don't really understand why they even let you put them there)

File: src/Facade/Services/PurchaseOrderFacadeService.cs [hunk 2]
```diff
@@ -97,6 +97,27 @@ public class PurchaseOrderFacadeService :
             }
         }
 
+        public async Task<IResult<ProcessResultResource>> EmailSupplierAss(int orderNumber)
+        {
+            try
+            {
+                var order = this.orderRepository.FindById(orderNumber);
+
+
```
Comment (lewisrenfrew) [inline src/Facade/Services/PurchaseOrderFacadeService.cs:106]: blankys

File: src/Facade/Services/PurchaseOrderFacadeService.cs [hunk 3]
```diff
@@ -97,6 +97,27 @@ public class PurchaseOrderFacadeService :
             }
         }
 
+        public async Task<IResult<ProcessResultResource>> EmailSupplierAss(int orderNumber)
```
Comment (lewisrenfrew) [inline src/Facade/Services/PurchaseOrderFacadeService.cs:100]: Does this need to be async (and return a task for that matter?) Don't see any await keyword in here

File: src/Persistence.LinnApps/Repositories/BomDetailRepository.cs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        }
+
+        public IQueryable<BomDetail> FindAll()
+        {
+            throw new NotImplementedException();
+        }
+
+        public IEnumerable<BomDetail> GetLiveBomDetails(string bomName)
+        {
+            return this.bomDetails.Include(b => b.Part).Where(b => b.BomName == bomName && b.ChangeState == "LIVE");
```
Comment (lewisrenfrew) [inline src/Persistence.LinnApps/Repositories/BomDetailRepository.cs:40]: Is there some reason why you couldn't just achieve this data in the usual FilterBy() method?

File: src/Service.Host/client/src/components/PurchaseOrders/PurchaseOrderUtility.js [hunk 5]
```diff
@@ -291,6 +292,15 @@ function PurchaseOrderUtility({ creating }) {
         );
     };
 
+    const handleSupplierAssEmailClick = () => {
+        reduxDispatch(sendPurchaseOrderSupplierAssActionTypes.clearProcessData);
```
Comment (lewisrenfrew) [inline src/Service.Host/client/src/components/PurchaseOrders/PurchaseOrderUtility.js:296]: did you mean to call clearProcessData()? If so there () is missing

File: src/Service.Host/client/src/components/PurchaseOrders/PurchaseOrderUtility.js [hunk 6]
```diff
@@ -587,6 +596,22 @@ function PurchaseOrderUtility({ creating }) {
                                 disabled
                             />
                         </Grid>
+                        <Grid item xs={1}>
+                            {order.issuePartsToSupplier === 'Y' ? (
```
Comment (lewisrenfrew) [inline src/Service.Host/client/src/components/PurchaseOrders/PurchaseOrderUtility.js:600]: I prefer the

 { order.issuePartsToSupplier === 'Y' && <Markup /> } 

syntax (over the ternary/conditional operator syntax) for these simple 'if something then render something ' style expressions. Just slightly less code - don't feel too strongly about it. s'up to you

**Review by lewisrenfrew — APPROVED** [review]

File: src/Domain.LinnApps/PurchaseOrders/ISupplierKitService.cs (hunk 1 above)
Comment (andrewfraser73) [reply src/Domain.LinnApps/PurchaseOrders/ISupplierKitService.cs:7]: all the other interfaces in that folder have them and if you take them it complains that it wants an access modifier internal or public

Comment (richardiphillips) [reply src/Domain.LinnApps/PurchaseOrders/ISupplierKitService.cs:7]: that can't be right. the public on line 7 should be removable?

File: src/Persistence.LinnApps/Repositories/BomDetailRepository.cs (hunk 4 above)
Comment (andrewfraser73) [reply src/Persistence.LinnApps/Repositories/BomDetailRepository.cs:40]: if i leave it to the usual filterby method then everyone is going to have to remember to always include the changestate check while this is the query folk are going to use 95% of the time.

Comment (richardiphillips) [reply src/Persistence.LinnApps/Repositories/BomDetailRepository.cs:40]: you can inherit the EntityFrameworkRepository and the interface and just implement / override the ones you need

File: src/Domain.LinnApps/PurchaseOrders/ISupplierKitService.cs (hunk 1 above)
Comment (andrewfraser73) [reply src/Domain.LinnApps/PurchaseOrders/ISupplierKitService.cs:7]: ah i misunderstood i thought you meant the public on the interface

**Pull request merged**